core/lib/multivm/src/versions/testonly/v26_upgrade_utils.rs

```diff
@@ -0,0 +1,289 @@
+        .with_empty_in_memory_storage()
+        .with_execution_mode(TxExecutionMode::VerifyExecute)
+        .with_custom_contracts(vec![ContractToDeploy::new(
+            test_contract.bytecode.to_vec(),
+            test_address,
+        )])
+        .with_rich_accounts(1)
+        .build::<VM>();
+
+    let system_tx = get_prepare_system_tx(
+        l1_chain_id,
+        l1_token_address,
+        l1_shared_bridge_address,
+        test_address,
+        test_contract,
+    );
+
+    vm.vm.push_transaction(system_tx);
+    let result = vm.vm.execute(InspectExecutionMode::OneTx);
+
+    assert!(!result.result.is_failed());
+
+    let l2_token_address = vm.storage.borrow_mut().read_value(&StorageKey::new(
+        AccountTreeId::new(COMPLEX_UPGRADER_ADDRESS),
+        H256::from_low_u64_be(0),
+    ));
+    let l2_legacy_shared_bridge_address = vm.storage.borrow_mut().read_value(&StorageKey::new(
+        AccountTreeId::new(COMPLEX_UPGRADER_ADDRESS),
+        H256::from_low_u64_be(1),
+    ));
+    let l1_aliased_shared_bridge = vm.storage.borrow_mut().read_value(&StorageKey::new(
+        AccountTreeId::new(COMPLEX_UPGRADER_ADDRESS),
+        H256::from_low_u64_be(2),
+    ));
+
+    let test_data = V26TestData {
+        l1_chain_id,
+        l1_shared_bridge_address,
+        l1_token_address,
+        l2_token_address: h256_to_address(&l2_token_address),
+        l2_legacy_shared_bridge_address: h256_to_address(&l2_legacy_shared_bridge_address),
+        l1_aliased_shared_bridge: h256_to_address(&l1_aliased_shared_bridge),
+    };
+
+    (vm, test_data)
+}
+
+fn encode_regisration(l2_token_address: Address) -> Vec<u8> {
+    let contract = l2_native_token_vault();
+
+    contract
+        .function("setLegacyTokenAssetId")
+        .unwrap()
+        .encode_input(&[Token::Address(l2_token_address)])
+        .unwrap()
+}
+
+// Returns a list of irrelevant keys that are not required to be the same.
+// For some reason they do not consistently reproduce between test runs, but they are not relevant to the
+// essence of the test.
+fn get_irrelevant_keys() -> Vec<StorageKey> {
+    vec![
+        StorageKey::new(AccountTreeId::new(L1_MESSENGER_ADDRESS), H256::zero()),
+        StorageKey::new(
+            AccountTreeId::new(SYSTEM_CONTEXT_ADDRESS),
+            H256::from_low_u64_be(10),
+        ),
+    ]
+}
+
+fn remove_irrelevant_keys(mut logs: HashMap<StorageKey, H256>) -> HashMap<StorageKey, H256> {
+    for to_remove in get_irrelevant_keys() {
+        logs.remove(&to_remove);
+    }
+
+    logs
+}
+
+pub(crate) fn test_trivial_test_storage_logs<VM: TestedVm>() {
+    let (vm, test_data) = setup_v26_unsafe_deposits_detection::<VM>();
+    assert_eq!(test_data, get_test_data());
+
+    let storage_ptr = vm.storage.clone();
+    let borrowed = storage_ptr.borrow();
+
+    let expected = remove_irrelevant_keys(borrowed.modified_storage_keys().clone());
+    let found = remove_irrelevant_keys(trivial_test_storage_logs());
```

Comment:
nit: shouldn't it be vice verse i.e. `expected` logs come from file and `found` from the actual execution?
Same for other tests

Reply:
I would say that no, the `expected` is the source of truth (i.e. how the execution really happens), while `found` is the JSON file that tracks the data.